cmd/router/main.go

```diff
@@ -104,40 +104,44 @@ type EnsembleStepOutput struct {
 	StepStatusCode int
 }
 
+// See if reviewer suggests a better name for this function
+func handleSplitterORSwitchNode(route *v1alpha1.InferenceStep, graph v1alpha1.InferenceGraphSpec, input []byte, headers http.Header) ([]byte, int, error) {
```

Comment:
Why it is only applying to splitter or switch node ? The picked route can be any node type imo

Reply:
Sorry for the confusing name of the function. The intent here was ti pull out the common code that was repeating in Splitter and Switch `if` sections.

Reply:
If you look at `if-else` branches for `Sequence` and `Ensemble`, they already have the handling there. I can refactor more in future PRs.